File: win-prog/ProgWinVB/Hello Windows Forms/PaintHello/PaintHello.vb
'-------------------------------------------
' PaintHello.vb (c) 2002 by Charles Petzold
'-------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Module PaintHello
    Sub Main()
        Dim frm As New Form()
        frm.Text = "Paint Hello"
        frm.BackColor = Color.White
        AddHandler frm.Paint, AddressOf MyPaintHandler

        Application.Run(frm)
    End Sub

    Sub MyPaintHandler(ByVal obj As Object, ByVal pea As PaintEventArgs)
        Dim frm As Form = DirectCast(obj, Form)
        Dim grfx As Graphics = pea.Graphics

        grfx.DrawString("Hello, world!", frm.Font, Brushes.Black, 0, 0)
    End Sub
End Module

File: win-prog/ProgWinCS/Hello Windows Forms/PaintHello/PaintHello.cs
//-----------------------------------------
// PaintHello.cs © 2001 by Charles Petzold
//-----------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class PaintHello
{
     public static void Main()
     {
          Form form      = new Form();
          form.Text      = "Paint Hello";
          form.BackColor = Color.White;
          form.Paint    += new PaintEventHandler(MyPaintHandler);

          Application.Run(form);
     }
     static void MyPaintHandler(object objSender, PaintEventArgs pea)
     {
          Form     form = (Form)objSender;
          Graphics grfx = pea.Graphics;

          grfx.DrawString("Hello, world!", form.Font, Brushes.Black, 0, 0);
     }
}